### PR TITLE
pythonPackages.autoflake: init at 1.0

### PR DESCRIPTION
**Pull request by yuriaisaka**
###### Motivation for this change

`autoflake` ( https://pypi.python.org/pypi/autoflake ) is a tool that automatically removes unused imports and unused variables from Python code.

Primarily, it is a standalone tool, but we put it under `python-modules` following the way how the similar package `pythonPackages.autopep8` is packaged.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/autoflake/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{ stdenv, buildPythonPackage, fetchPypi, pyflakes }:
+
+buildPythonPackage rec {
```
Comment (Mic92) [inline pkgs/development/python-modules/autoflake/default.nix]: This does not look a library to me. If so, can you use `buildPythonApplication` and move it to `all-packages.nix`?

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -661,6 +661,8 @@ in {
     };
   };
 
+  autoflake = callPackage ../development/python-modules/autoflake { };
+
   autopep8 = callPackage ../development/python-modules/autopep8 { };
```
Comment (Mic92) [inline pkgs/top-level/python-packages.nix]: `autopep8` might be also a candidate... but this does not need to be done in this pr.

**Comment by yuriaisaka** [comment]
@Mic92 Thank you very much for the suggestions, I agree that this (and perhaps also `autopep8`) fit better in `all-packages.nix`. (Ping @bjornfor ?)

Anyways, I've updated the pr and made it a Python3 app.
Hope now it conforms to https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.md

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -511,6 +511,8 @@ with pkgs;
     python3Packages = python34Packages;
   };
 
+  autoflake = callPackage ../development/python-modules/autoflake { };
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: maybe:

```
pkgs/development/tools/analysis/
```

as a directory?

Comment (yuriaisaka) [reply pkgs/top-level/all-packages.nix]: Yep, I'll do that, thanks!

Or, maybe can I dig `pkgs/development/tools/python/` and use the directory?
Langs like `guile`, `haskell`, `ocaml`, `java`, .. seem to be doing that.
`go` is a notable exception and spreads `goimports`, `gotags` etc right under `developement/tools`.

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**